### PR TITLE
Enable or disable bf16 support based on availability

### PR DESCRIPTION
**Pull request by simhallq**
Issue #1090 

Added support for `auto` in bf16 setting of config-files. Now sets bf16 to True/False depending on GPU-support if `bf16: auto`.

### CODE REVIEW

**Review by winglian — APPROVED** [review]

File: src/axolotl/utils/config.py [hunk 1]
```diff
@@ -61,6 +61,14 @@ def normalize_config(cfg):
         cfg.device_map = {"": int(os.environ.get("LOCAL_RANK", 0))}
         cfg.batch_size = cfg.batch_size * cfg.world_size
 
+    if cfg.bf16 == "auto":
+        if is_torch_bf16_gpu_available():
+            LOG.info("bf16 support detected, enabling for this configuration.")
```
Comment (winglian) [inline src/axolotl/utils/config.py]: Can we change the visibility of these logs to debug level please? Thanks!

**Comment by winglian** [comment]
Also if you're feeling ambitious, adding a test for normalize config that it sets bf16 appropriately based on a mock of is_torch_bf16_gpu_available.

**Comment by simhallq** [comment]
@winglian I've changed logging level and added a unit test.

**Comment by simhallq** [comment]
Btw the test fails on Mac since `device=="mps"` sets `bf16==False` (which is the desired behavior), should I mock the value of `choose_device` in the test or is that overkill?

**Review by winglian — APPROVED** [review]
Perfect. Thanks! No need to worry about mocking the device for now.

**Comment by simhallq** [comment]
@winglian I just realized I missed something - `validate_config` now throws an error at [this](https://github.com/OpenAccess-AI-Collective/axolotl/blob/c1b741d9fb51c6e2d5bcda960524bbdd6fc21bf7/src/axolotl/utils/config.py#L186 ) line if bf16 is unavailable, since we are not explicility checking `cfg.bf16 is True` but just checking for any truthy value (which includes non-empty string like "auto").

Should I open a new PR with the fix?

**Pull request merged**